src/engraving/playback/renderers/chordarticulationsrenderer.cpp

```diff
@@ -54,14 +60,48 @@ void ChordArticulationsRenderer::doRender(const EngravingItem* item, const mpe::
         return;
     }
 
-    const Chord* chord = toChord(item);
+    for (const auto& type : ctx.commonArticulations) {
+        if (muse::contains(GRACE_NOTE_ARTICULATION_TYPES, type.first)) {
```

Comment:
Another possible simplification: instead of doing grace notes detection via "articulations" (which is a bit weird from a musical perspective, because grace notes are not really articulations in the same sense as things like staccato, tenuto, accent), just check directly whether the `Chord` has grace chords. 
However this could be done in a separate PR.

Reply:
>just check directly whether the Chord has grace chords

This would indeed be better; however, we also need to know what type of grace chords is used (which is already stored in commonArticulations). So that's the simplest way to do it at the moment